sound/soc/sof/ipc4-fw-reg.h

```diff
@@ -0,0 +1,139 @@
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ */
+
+#ifndef __IPC4_FW_REG_H__
+#define __IPC4_FW_REG_H__
+
+#define SOF_IPC4_INVALID_STREAM_POSITION	ULLONG_MAX
+
+struct sof_ipc4_pipeline_registers {
+	/*
+	 * Stream start offset (LPIB) reported by mixin module allocated on pipeline attached
+	 * to Host Output Gateway when first data is being mixed to mixout module. When data
+	 * is not mixed (right after creation/after reset) value "(u64)-1" is reported.
+	 * In number of bytes.
+	 */
+	u64 stream_start_offset;
+	/*
+	 * Stream end offset (LPIB) reported by mixin module allocated on pipeline attached
+	 * to Host Output Gateway during transition from RUNNING to PAUSED. When data
+	 * is not mixed (right after creation/after reset) value "(u64)-1" is reported. When
+	 * first data is mixed then value "0"is reported.
+	 * In number of bytes.
+	 */
+	u64 stream_end_offset;
+} __packed __aligned(4);
+
+#define SOF_IPC4_PV_MAX_SUPPORTED_CHANNELS 8
+
+struct sof_ipc4_peak_volume_regs {
+	u32 peak_meter[SOF_IPC4_PV_MAX_SUPPORTED_CHANNELS];
+	u32 current_volume[SOF_IPC4_PV_MAX_SUPPORTED_CHANNELS];
+	u32 target_volume[SOF_IPC4_PV_MAX_SUPPORTED_CHANNELS];
+} __packed __aligned(4);
+
+struct sof_ipc4_llp_reading {
+	/* lower part of 64-bit LLP */
+	u32 llp_l;
+	/* upper part of 64-bit LLP */
+	u32 llp_u;
+	/* lower part of 64-bit Wallclock */
+	u32 wclk_l;
+	/* upper part of 64-bit Wallclock */
+	u32 wclk_u;
+} __packed __aligned(4);
+
+struct sof_ipc4_llp_reading_extended {
+	struct sof_ipc4_llp_reading llp_reading;
+	/* total processed data (low part) */
+	u32 tpd_low;
+	/* total processed data (high part) */
+	u32 tpd_high;
+} __packed __aligned(4);
+
+struct sof_ipc4_llp_reading_slot {
+	u32 node_id;
+	struct sof_ipc4_llp_reading reading;
+} __packed __aligned(4);
+
+/* ROM info */
+#define SOF_IPC4_FW_FUSE_VALUE_MASK		GENMASK(7, 0)
+#define SOF_IPC4_FW_LOAD_METHOD_MASK		BIT(8)
+#define SOF_IPC4_FW_DOWNLINK_IPC_USE_DMA_MASK	BIT(9)
+#define SOF_IPC4_FW_LOAD_METHOD_REV_MASK	GENMASK(11, 10)
+#define SOF_IPC4_FW_REVISION_MIN_MASK		GENMASK(15, 12)
+#define SOF_IPC4_FW_REVISION_MAJ_MASK		GENMASK(19, 16)
+#define SOF_IPC4_FW_VERSION_MIN_MASK		GENMASK(23, 20)
+#define SOF_IPC4_FW_VERSION_MAJ_MASK		GENMASK(27, 24)
+
+/* Number of pipeline registers slots in FW Regs. */
+#define SOF_IPC4_MAX_SUPPORTED_ADSP_CORES 8
+
+/* Number of host pipeline registers slots in FW Regs. */
+#define SOF_IPC4_MAX_PIPELINE_REG_SLOTS 16
+
+/* Number of PeakVol registers slots in FW Regs. */
+#define SOF_IPC4_MAX_PEAK_VOL_REG_SLOTS 16
+
+/* Number of GPDMA LLP Reading slots in FW Regs. */
+#define SOF_IPC4_MAX_LLP_GPDMA_READING_SLOTS 24
+
+/* Number of Aggregated SNDW Reading slots in FW Regs. */
+#define SOF_IPC4_MAX_LLP_SNDW_READING_SLOTS 16
+
+/* Current ABI version of the FwRegisters layout. */
+#define SOF_IPC4_FW_REGS_ABI_VER 1
```

Comment:
is this something you defined yourself, or something that the firmware also knows about?

Reply:
it is copied from SOF FW